packages/cli/src/ui/commands/ideCommand.ts

```diff
@@ -4,40 +4,27 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { fileURLToPath } from 'url';
-import { Config, IDEConnectionStatus } from '@google/gemini-cli-core';
+import {
+  Config,
+  getIdeDisplayName,
+  getIdeInstaller,
+  IDEConnectionStatus,
+} from '@google/gemini-cli-core';
 import {
   CommandContext,
   SlashCommand,
   SlashCommandActionReturn,
   CommandKind,
 } from './types.js';
-import * as child_process from 'child_process';
-import * as process from 'process';
-import { glob } from 'glob';
-import * as path from 'path';
-
-const VSCODE_COMMAND = process.platform === 'win32' ? 'code.cmd' : 'code';
-const VSCODE_COMPANION_EXTENSION_FOLDER = 'vscode-ide-companion';
-
-function isVSCodeInstalled(): boolean {
-  try {
-    child_process.execSync(
-      process.platform === 'win32'
-        ? `where.exe ${VSCODE_COMMAND}`
-        : `command -v ${VSCODE_COMMAND}`,
-      { stdio: 'ignore' },
-    );
-    return true;
-  } catch {
-    return false;
-  }
-}
 
 export const ideCommand = (config: Config | null): SlashCommand | null => {
   if (!config?.getIdeMode()) {
     return null;
   }
+  const currentIDE = config.getIdeClient()?.getCurrentIde();
+  if (!currentIDE) {
+    return null;
```

Comment:
This will trigger if we don't recognize the IDE right? Should we do a proper error here?